metricbeat/module/kubernetes/event/event.go

```diff
@@ -193,8 +205,14 @@ func generateMapStrFromEvent(eve *kubernetes.Event, dedotConfig dedotConfig) com
 
 	tsMap := make(common.MapStr)
 
-	tsMap["first_occurrence"] = kubernetes.Time(&eve.FirstTimestamp).UTC()
-	tsMap["last_occurrence"] = kubernetes.Time(&eve.LastTimestamp).UTC()
+	lastTimestamp := kubernetes.EventLastTimestamp(eve)
+	tsMap["first_occurrence"] = func() time.Time {
+		if eve.FirstTimestamp.IsZero() {
+			return kubernetes.Time(&lastTimestamp).UTC()
```

Comment:
I'm not sure if this is correct. If an event has no `FirstTimestamp` attached then we fallback to `lastTimestamp` so this means that each time the event occurs we emit data with different `first_occurrence` which is always equal to the current `lastTimestamp`. Is this something that is intentional, and if so why?

Maybe it's better here to use `EventTime` as a fallback?